### PR TITLE
Add concept

### PR DESCRIPTION
**Pull request by dianaboiangiu**

### CODE REVIEW

File: gemet/thesaurus/models.py [hunk 1]
```diff
@@ -25,6 +25,13 @@
     'seeAlso',
 ]
 
+EDIT_URL_NAMES = {
```
Comment (iuliachiriac) [inline gemet/thesaurus/models.py]: Move this to `edit_views.py` or `definitions.py`. If you choose the latter, move all constants to this new file.

File: gemet/thesaurus/edit_views.py [hunk 2]
```diff
@@ -429,3 +431,44 @@ def post(self, request, langcode, id, relation_id):
         data = {'restore_url': restore_url}
 
         return self._get_response(data, 'success', 200)
+
+
+class AddNewConceptView(View):
```
Comment (iuliachiriac) [inline gemet/thesaurus/edit_views.py]: Keep it simple: `AddConceptView`

File: gemet/thesaurus/edit_views.py [hunk 3]
```diff
@@ -429,3 +431,44 @@ def post(self, request, langcode, id, relation_id):
...
+        context = {
+            'language': Language.objects.get(code=langcode),
+            'form': form
+        }
+        return render(request, 'edit/concept_add.html', context)
+
+    def post(self, request, langcode):
+        language = Language.objects.get(code=langcode)
+        form = ConceptForm(request.POST)
+        import pdb;pdb.set_trace()
```
Comment (iuliachiriac) [inline gemet/thesaurus/edit_views.py]: Shame :bell: shame :bell: shame :bell:

File: gemet/thesaurus/edit_views.py [hunk 4]
```diff
@@ -429,3 +432,43 @@ def post(self, request, langcode, id, relation_id):
...
         return self._get_response(data, 'success', 200)
+
+
+class AddConceptView(View):
+
+    def get(self, request, langcode):
+        form = ConceptForm()
+        context = {
+            'language': Language.objects.get(code=langcode),
+            'form': form
```
Comment (iuliachiriac) [inline gemet/thesaurus/edit_views.py]: Comma after the last element in an iterable.

File: gemet/thesaurus/edit_views.py [hunk 5]
```diff
@@ -429,3 +432,43 @@ def post(self, request, langcode, id, relation_id):
...
+        }
+        return render(request, 'edit/concept_add.html', context)
+
+    def post(self, request, langcode):
+        language = Language.objects.get(code=langcode)
+        form = ConceptForm(request.POST)
+        if form.is_valid():
+            version = Version.objects.create()
+            namespace = form.cleaned_data['namespace']
+            new_concept = Term(version_added=version,
```
Comment (iuliachiriac) [inline gemet/thesaurus/edit_views.py]: `Term` implies that the namespace is `'Concepts'`.

File: gemet/thesaurus/templates/edit/concept_add.html [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+{% endblock %}
+
+{% block content %}
+
+  <h1> Create new concept </h1>
+
+  <form action="{% url 'concept_add' language.code %}" method="POST" class="post-form">
+    {% csrf_token %}
+    {{ form.as_p }}
+    <button type="submit" value="Submit">Submit</button>
```
Comment (iuliachiriac) [inline gemet/thesaurus/templates/edit/concept_add.html]: I think `Save` is more suitable here than `Submit`.

**Comment by dianaboiangiu** [comment]
Requested changes made.

File: gemet/thesaurus/edit_views.py [hunk 7]
```diff
@@ -429,3 +432,44 @@ def post(self, request, langcode, id, relation_id):
...
+        }
+        return render(request, 'edit/concept_add.html', context)
+
+    def post(self, request, langcode):
+        language = Language.objects.get(code=langcode)
+        form = ConceptForm(request.POST)
+        if form.is_valid():
+            version = Version.objects.create()
+            namespace = form.cleaned_data['namespace']
+            Concept_class = CONCEPT_TYPES[namespace.heading]
```
Comment (iuliachiriac) [inline gemet/thesaurus/edit_views.py]: This doesn't look right. Why aren't you using the `Concept` class?

File: gemet/thesaurus/templates/edit/concept_add.html [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+{% endblock %}
+
+{% block content %}
+
+  <h1> Create new concept </h1>
+
+  <form action="{% url 'concept_add' language.code %}" method="POST" class="post-form">
+    {% csrf_token %}
+    {{ form.as_p }}
+    <button type="submit" value="Save">Submit</button>
```
Comment (iuliachiriac) [inline gemet/thesaurus/templates/edit/concept_add.html]: You did't modify the text that is actually visible on the button.

**Pull request merged**